dev/build.go

```diff
@@ -0,0 +1,83 @@
+package dev
+
+import (
+	"fmt"
+	"github.com/magefile/mage/mg"
+	"github.com/magefile/mage/sh"
+	"log"
+	"os"
+	"os/exec"
+)
+
+type ImageBuildInfo struct {
+	ImageTag      string
+	CacheDir      string
+	ContainerFile string
+	ContextDir    string
+	Runtime       string
+}
+
+type ImagePushInfo struct {
+	DigestFile string
+	BuildInfo  *ImageBuildInfo
+}
+
+func must(err error) {
+	if err != nil {
+		panic(err)
+	}
+}
+
+// BuildImage is a generic image build function,
+// requires the binaries to be built beforehand
+func BuildImage(buildInfo *ImageBuildInfo, deps []interface{}) {
```

Comment:
Is it possible to be more specific that `interface{}`

Reply:
Unfortunately not, the `mg.SerialDeps()` function takes arguments of type `interface{}`